src/strategy/immediate.rs

```diff
@@ -31,6 +31,7 @@ impl StrategyImmediate {
     pub(crate) fn report_steady(&self) -> Pin<Box<dyn Future<Output = Result<bool, Error>>>> {
         trace!("immediate strategy, report steady: {}", true);
 
+        // TODO(zonggen): https://github.com/coreos/zincati/issues/35
```

Comment:
You can drop this TODO line. Additionally, I think both the `check` and `finalize` fields on this struct can be removed, as I think they are anyway always set to `true` (they were initially supposed to be configurable, but we dropped that part during initial review).

Reply:
Dropped both `check` and `finalize`, and added a simple test for `can_finalize`.